python/ray/serve/_private/http_proxy.py

```diff
@@ -322,6 +337,14 @@ async def __call__(self, scope, receive, send):
                 scope, receive, send
             )
 
+        # The current autoscale logic can downscale nodes with ongoing requests if the
+        # node doesn't have replicas and has no object references. This counter and
+        # the dummy object reference will have to keep the node alive while draining
+        # requests, so they are not dropped unintentionally.
+        self._ongoing_requests += 1
+        if self._ongoing_requests > 0 and self._ongoing_requests_dummy_obj_ref is None:
+            self._ongoing_requests_dummy_obj_ref = ray.put("ongoing_requests")
```

Comment:
I just realized– one downside of this approach is that `ray.put` is now on the critical path for the data plane. If the GCS is down, I believe this call will hang. @edoakes any ideas how to get around this?

Reply:
Synced with @edoakes and @GeneDer offline. `ray.put` shouldn't hang even while the GCS is down.

Reply:
Talked offline, `ray.put` shouldn't hang and GCS is not required.